### DISCUSSION TITLE
How to control warm water ciculation pump?

### DISCUSSION THREAD

**Discussion by RosemannBluro** [emsesp/EMS-ESP32#1766] [Q&A]
Hey guys,

since a couple of month we use a Buderus WLW196i and EMS-ESP Gateway (EMS-ESP Version 3.6.4) works like a charme. I can control pv surplus to activate thermal buffer, change the target temperature and everything. But there is one thing, that i cant control and that is the circulation pump.

`WW Zirkulationspumpe vorhanden` is active

If i want to change `WW Zirkulation aktiv` the value sets back to the setting before and nothing happen. 
If i change `WW Zirkulationspumpen-Modus` to `off` it sets to the next minimum `1x 3 Min`.
Is there anything i have to change to control the circulation pump by my own?
I would like to disable ciculation during the night or i nobody is at home

Thanks in advanced
Luca

**Comment by proddy** [comment]
let's get the wwCircMode working first on your setup. We'll need the system info (https://emsesp.github.io/docs/Support/) and also the log (https://emsesp.github.io/docs/Troubleshooting/#changing-a-value-on-an-ems-device-doesnt-work)

**Reply by proddy** [reply]
and I would update to the latest dev version 3.7.0

**Comment by RosemannBluro** [comment]
I have an update for everyone who is struggling with the same problem.
@proddy i dont think, that the EMS itself is the problem. I figured out, that you cant control the switch `WW Zirkulation aktiv` directly. The heat pump dont accept changes. This is more like a sensor to read the current status.

But i was able to manage the control with the switch `WW Zirkulationspumpe vorhanden` in combination with `WW Zirkulationspumpen-Modus`

**Comment by proddy** [comment]
wwCirc (WW Zirkulation aktiv) should enable/disable the pump on a boiler. We need to adjust the code for your heatpump if you could share the logs.

I'm glad it works with wwCircPump (WW Zirkulationspumpe vorhanden) and wwCircMode (Zirkulationspumpen-Modus) so thanks for sharing that information. Do note in the next version (3.7.0) the 'WW' has been replaced with 'DHW'

**Comment by bergteam** [comment]
Same Problem here, I would like to control Circulation Pump (on/off) from outside. Having a Junkers zsb 14-5.2 c 23.

1. "dhw circulation pump available" tells the boiler a pump is available
2. "dhw circulation pump mode" sets the "number of runs per hour", the switching off and contineous settings from EMS-ESP don't work. This is in line with the Thermostat FW120 settings, were at this parameter only the number of times running can be selected.
3. My assumption is, that the Thermostat switches the pump on/off via the configured schedule

=> There must be a diagram on the bus switching the circualation pump on the boiler (on/off). Note regardless if the pump runs or not, the "dhw circulation pump available" setting is always "on". Does it help to log all diagrams around the time the scheduler (of the Thermostat) switches the pump on or off?

**Reply by wisniole** [reply]
Set the following parameters permanently:
[DHW circulation pump mode]=continuous (1x3, 2x3 possible too),
[DHW circulation active]=OFF.
The circulation pump is controlled by:
[dhw circulation pump available] = ON/OFF.
When set to ON, the pump turns on ([DHW circulation active] changes to ON), when set to OFF, the pump turns off ([DHW circulation active] changes to OFF). I wait 10 seconds for the pump operating status to change.
I have now set it in EMS-ESP, I don't know how the boiler behaves, I will check it when I get home.
Buderus Logamax Plus GB 072-14

**Reply by bergteam** [reply]
Basically I agree. There are 3 parameter with my "CBS/GB*72, Trendline, Cerapur, Greenstar Si (Boiler)".
+ dhw circulation active
    //although writeable this changes to "on" once the therostat switches the pump on
+ dhw circulation pump available
    //a configuration setting, telling the boiler there is a pump **physically** available
+ dhw circulation pump mode
    //change the mode between x-times per hour setting, the available "off" value won't work

=> I did not dare to control the pump (afraid of side effects) via "dhw circulation pump available" as this parameter physically tells the boiler a pump is **installed** or not.

**Reply by wisniole** [reply]
Exactly.
You can't turn on the pump if it's "not installed" ;)
Once you "install" the pump, only then can you start the circulation. Basically, the circulation starts immediately (in my case after 10 seconds) when the pump is connected, according to the parameter set in [DHW circulation pump mode].
I will soon implement it in my Home Assistant and control it using motion sensors in the bathrooms. The circulation will be activated when there is a potential demand for hot water.
I control the boiler using Honeywell Evohome, so I cannot use the native Buderus room controller (e.g. RC-310).

**Comment by wisniol** [comment]
I implemented my idea of ​​controlling the circulation pump using a motion sensor. Works great.
1. After detecting motion, I run: [dhw circulation pump avalible]=ON;
   -- boiler automatically set [dhw circulation active] = [ON] for 3 minutes;
   -- In HA, TIMER = 19'30'' starts, i.e. it ends before the next circulation cycle starts;
2. When the timer ends, the [dhw circulation pump avalible]=OFF setting is triggered.
When someone enters the bathroom again, the above automation starts again.
![image](https://github.com/user-attachments/assets/6073b678-2680-4d8e-b6df-9ec2b3bb6802)
![image](https://github.com/user-attachments/assets/773e09b6-00eb-46d6-a361-e1804246ad9b)

**Comment by sholl** [comment]
I tried to reproduce your process given in [this comment](https://github.com/emsesp/EMS-ESP32/discussions/1766#discussioncomment-11716556) on my maschine Buderus CS*800i, Logatherm WLW\* (Boiler/HP), connected with EMS-ESP (3.7.2) which does not work actually. Any hint where to debug? Setting the state of the switch "switch.boiler_dhw_circulation_active" from within the Developer Tools State-Tab inside Homeassistant enables is though, but no switch.turn-on-action from within an automation.
Any hints where do dig in?
